### DISCUSSION TITLE
MPI compiling on Gricad

### DISCUSSION THREAD

**Discussion by vittofra** [nextsimdg/nextsimdg#702] [General]
Hello @auraoupa @stephanieleroux, 
I am trying to compile the latest version of the develop branch of nextsimdg on Gricad. 
When entering the cmake call, I ask to enable MPI support as follows: cmake .. -DENABLE_MPI=ON
However, I get the following issue: 
![immagine](https://github.com/user-attachments/assets/0da9e47f-22c8-4f84-8e7a-f7c90647bce6)
I think I should explicit the path where to find the mpicxx. But I do not know where it is on Gricad. 
Have you ever managed to compile with this option? 
Do you have any solution? 

Thanks

**Comment by auraoupa** [comment]
Hi @vittofra, we never tried to compile nextsimdg with MPI on GRICAD before, I got the same error than you while trying with the current conda environment. I could not find any suited conda library so I will try to implement the spack environment defined by @TomMelt, I'll keep you posted on how it goes !

**Comment by auraoupa** [comment]
Hello @vittofra, some updates : the installation with spack was not successful (see #705) so I continued my exploration with an apptainer image (docker is not available on most HPC), it compiled nicely but I still need to work on it for the running part. And then the part where the computing cores of GRICAD are used properly for the parallelization needs more work also (following https://gricad-doc.univ-grenoble-alpes.fr/en/hpc/softenv/container/)

All of this made me realize that maybe the MPI version of nextsimdg is not quite ready yet, you can see [here](https://github.com/nextsimhub/nextsimdg/issues/690) and [here](https://github.com/nextsimhub/nextsimdg/issues/689) that they are some issues left, so I wonder wether it is absolutely necessary for your work to run on multiple processors ? Can you elaborate why so we can find a workaround until all of this is sorted ?

**Reply by vittofra** [reply]
Hello @auraoupa, thanks for your feedback. 

At this stage (today) it is not strictly necessary for us to use the parallelised version. 

We wanted to test a recent version (which appears to have parallelisation) and try to exploit MPI to reduce execution time. 
While waiting for things to settle down, we can continue working without MPI. 

@ aydogduali missed anything?

**Comment by aydogduali** [comment]
@vittofra, I think you didn't miss anything for now. We will revisit this discussion at a later stage. Thanks @auraoupa
